enhancements/operator-sdk-builds-catalogs.md

```diff
@@ -0,0 +1,179 @@
+- Add any other functionality other than build a catalog locally (`opm index add`).
+
+## Proposal
+
+### User Stories
+
+#### Build a catalog using `opm`
+
+As an Operator developer, I want to easily build a catalog from my Operator's bundles
+without having to know the details of _how_ it is built.
+
+#### Understand what a catalog is
+
+As an Operator developer, I should understand what a catalog is and why I would consider
+building one. I also should understand how bundling and building a catalog ties into shipping my Operator.
+
+### Implementation Details/Notes/Constraints
+
+There are two ways of adding `opm index add` functionality to `operator-sdk`:
+1. Create a `catalog-build` Makefile rule that calls `opm index add`.
+1. Add a `operator-sdk catalog build` subcommand (discussed in the [alternatives](#subcommand-approach) section).
+
+The Makefile approach is by far the best for a few reasons:
+- Users have full control over which `opm` version to use.
+- Users have full control over defaults for their project, including image tags.
+- Users get experience using `opm`, but only the relevant part(s).
+  - They can easily add more `opm` targets if they want.
+
+The Makefile additions would look like:
+```make
+VERSION ?= 0.1.0
+IMAGE_TAG_BASE ?= quay.io/example/my-operator
+BUNDLE_IMG ?= $(IMAGE_TAG_BASE)-bundle:$(VERSION)
+
+bundle-push:
+	$(MAKE) docker-push IMG=$(BUNDLE_IMG)
+
+OPM = ./bin/opm
+opm:
+  # Download `opm` from github.com/operator-framework/operator-registry/releases
+
+BUNDLE_IMGS ?= $(BUNDLE_IMG)
+CATALOG_IMG ?= $(IMAGE_TAG_BASE)-catalog:v$(VERSION)
+catalog-build: opm
+	$(OPM) index add --mode semver --tag $(CATALOG_IMG) --bundles $(BUNDLE_IMGS)
+
+catalog-push:
+	$(MAKE) docker-push IMG=$(CATALOG_IMG)
+```
+
+In addition, consider these changes to the current Makefile:
+
+```diff
+VERSION ?= 0.1.0
++IMAGE_TAG_BASE ?= quay.io/example/my-operator
+-BUNDLE_IMG ?= controller-bundle:$(VERSION)
++BUNDLE_IMG ?= $(IMAGE_TAG_BASE)-bundle:v$(VERSION)
+
++bundle-push:
++	$(MAKE) docker-push IMG=$(BUNDLE_IMG)
+```
+
+Coupled with the suggested `opm` additions, these changes would allow target chaining like so:
+
+```sh
+export IMAGE_TAG_BASE=quay.io/example/my-operator
+make bundle-build bundle-push catalog-build catalog-push
+```
+
+Resulting in both `quay.io/example/my-operator-bundle:v0.1.0` and `quay.io/example/my-operator-catalog:v0.1.0`
+being built and pushed.
+
+Check out this [PR](https://github.com/operator-framework/operator-sdk/pull/4406) for an implementation reference.
+
+### Risks and Mitigations
+
+The major risk, which has already been encountered, is an issue with an `opm` release.
+In particular, problems with building a binary on a particular architecture may result
+in no binary being released for that architecture. `operator-sdk` supports
+amd64 (for linux and darwin platforms), arm64, ppc64le, and s390x; a binary targeting each
+platform must exist for every `opm` release. Mitigation involves
```

Comment:
Can we put Mitigation involves on a new line? IMO it would make it easier to see the RISK vs the MITIGATION.